### PR TITLE
Init widget api

### PR DESCRIPTION
**Pull request by hblitza**
This initializes the widget api within the klips-sdi repo.  
It is a simple web application only using vite, typescript and echarts and parses the url location (search params or pathnames) to create a chart.

In follow-up pr:
- tests (!)
- connect klips-api to get data
- validation of parameters (e.g. geometry checking)

Peek 2023-05-12 18-58

@chrismayer @sdobbert please check

### CODE REVIEW

**Review by hwbllmnn — APPROVED** [review]

File: klips-widget-api/package.json [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    "email": "blitza@terrestris.de",
+    "url": "https://www.terrestris.de"
+  },
+  "version": "1.0.0",
+  "type": "module",
+  "scripts": {
+    "dev": "vite",
+    "build": "tsc && vite build",
+    "preview": "vite preview",
+    "typecheck": "tsc",
```
Comment (dnlkoch) [inline klips-widget-api/package.json:16]:
```suggestion
    "typecheck": "tsc --noEmit",
```

File: klips-widget-api/src/main.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+import './style.css';
+import { getParams, parseURLPathnames } from './util/Url';
+import { validateParams } from './util/Config.js';
+
+import ChartApi from './components/Chart/index.js';
+
+import { Params } from './types';
+import { pathNameConfig } from './constants/index.js';
+
+const widgetApi = class WidgetAPI {
```
Comment (dnlkoch) [inline klips-widget-api/src/main.ts:11]: Very minor, but the `WidgetAPI` can be moved to an own file/module.

File: klips-widget-api/src/components/Chart/index.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
+/* eslint-disable no-console */
+import { Params } from '../../types';
+
+import * as echarts from 'echarts';
+
+export type EChartsOption = echarts.EChartsOption;
+
+export const chartApi = class ChartAPI {
```
Comment (dnlkoch) [inline klips-widget-api/src/components/Chart/index.ts:8]:
```suggestion
export class ChartAPI {
```

File: klips-widget-api/src/main.ts (hunk 2 above)
Comment (dnlkoch) [inline klips-widget-api/src/main.ts:11]:
```suggestion
class WidgetAPI {
```

**Review by dnlkoch — APPROVED** [review]

File: klips-widget-api/src/main.ts (hunk 2 above)
Comment (hblitza) [reply klips-widget-api/src/main.ts]: yup, will do thanks

**Review by chrismayer — APPROVED** [review]
LGTM so far.

**Comment by hblitza** [comment]
Thanks!

**Pull request merged**